src/main/java/com/github/onsdigital/babbage/search/helpers/SearchHelper.java

```diff
@@ -61,14 +65,44 @@ public static ONSSearchResponse search(ONSQuery queries) {
 
         List<ONSSearchResponse> helpers = new ArrayList<>();
         MultiSearchResponse response = multiSearchRequestBuilder.get();
-        {
-            int i = 0;
-            for (MultiSearchResponse.Item item : response.getResponses()) {
-                if (item.isFailure()) {
-                    throw new ElasticsearchException(item.getFailureMessage());
+        MultiSearchResponse.Item[] items = response.getResponses();
+
+        for (int i = 0; i < items.length; i++) {
+            MultiSearchResponse.Item item = items[i];
+
+            if (item.isFailure()) {
+                throw new ElasticsearchException(item.getFailureMessage());
+            }
+
+            ONSQuery query = queries.get(i);
+            ONSSearchResponse searchResponse = resolveDetails(query, new ONSSearchResponse(item.getResponse()));
+
+            if (null != query.types()) {
+                List<ContentType> types = Arrays.asList(query.types());
+                // Check for retired product pages
+                if (query.size() == 1 &&
+                        types.contains(ContentType.product_page)) {
+                    // Get the hit - we only queried for 1 document so we should only get 1 back
+                    SearchHits searchHits = searchResponse.response.getHits();
+                    if (searchHits.getHits().length == 1) {
+                        SearchHit searchHit = searchHits.getAt(0);
+                        // ID is the url
+                        String url = searchHit.getId();
+
+                        if (!Configuration.ELASTIC_SEARCH.getHighlightBlacklist().contains(url)) {
```

Comment:
Not sure its necessary but do you need to format the url - trailing / etc? If one has it and ones doesn't this wont match it.